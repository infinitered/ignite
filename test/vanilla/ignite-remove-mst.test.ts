import { filesystem } from "gluegun"
import * as tempy from "tempy"
import { runIgnite } from "../_test-helpers"

const BOILERPLATE_PATH = filesystem.path(__dirname, "../../boilerplate")

const setup = (): { TEMP_DIR: string } => {
  const TEMP_DIR = tempy.directory({ prefix: "ignite-" })

  beforeEach(() => {
    // create the destination directory
    filesystem.dir(TEMP_DIR)
    // copy the relevant files & folders
    filesystem.copy(BOILERPLATE_PATH + "/package.json", TEMP_DIR + "/package.json", {
      overwrite: true,
    })
    filesystem.copy(BOILERPLATE_PATH + "/app", TEMP_DIR + "/app", { overwrite: true })
  })

  afterEach(() => {
    filesystem.remove(TEMP_DIR) // clean up our mess
  })

  return { TEMP_DIR }
}

describe("ignite-cli remove-mst", () => {
  const { TEMP_DIR } = setup()

  it("should print the expected response", async () => {
    const result = await runIgnite(`remove-mst ${TEMP_DIR}`)

    // "/user/home/ignite" replaces the temp directory, so we don't get failures when it changes every test run
    const MOCK_DIR = `/user/home/ignite`
    const output = result.replace(new RegExp(TEMP_DIR, "g"), MOCK_DIR)

    expect(output).toMatchSnapshot()
  })
})
