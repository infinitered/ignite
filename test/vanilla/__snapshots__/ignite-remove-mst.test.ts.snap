// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ignite-cli remove-mst should print the expected response 1`] = `
"   
   Removing MobX-State-Tree code from '/user/home/ignite'
   Removing dependencies from package.json: mobx, mobx-react-lite, mobx-state-tree, reactotron-mst 
removing file /user/home/ignite/app/models/RootStore.ts
removing file /user/home/ignite/app/models/helpers/getRootStore.ts
removing file /user/home/ignite/app/models/helpers/setupRootStore.ts
removing file /user/home/ignite/app/models/helpers/useStores.ts
removing file /user/home/ignite/app/models/helpers/withSetPropAction.ts
removing file /user/home/ignite/app/models/index.ts
   Found '@mst remove-current-line', '@mst replace-next-line' in /user/home/ignite/app/app.tsx
   Found '@mst remove-current-line', '@mst remove-block-start', '@mst remove-block-end' in /user/home/ignite/app/devtools/ReactotronConfig.ts
   Found 'remove-file' in /user/home/ignite/app/models/helpers/getRootStore.ts
   Found 'remove-file' in /user/home/ignite/app/models/helpers/setupRootStore.ts
   Found 'remove-file' in /user/home/ignite/app/models/helpers/useStores.ts
   Found 'remove-file' in /user/home/ignite/app/models/helpers/withSetPropAction.ts
   Found 'remove-file' in /user/home/ignite/app/models/index.ts
   Found 'remove-file' in /user/home/ignite/app/models/RootStore.ts
   Found '@mst remove-current-line', '@mst replace-next-line' in /user/home/ignite/app/navigators/AppNavigator.tsx
   Found '@mst remove-current-line', '@mst replace-next-line' in /user/home/ignite/app/screens/WelcomeScreen.tsx
   Removed empty directory '/user/home/ignite/app/models/helpers'
   Done removing MobX-State-Tree code from '/user/home/ignite'
"
`;
